lib/models.ex

```diff
@@ -0,0 +1,20 @@
+defmodule Band do
+  @derive Jason.Encoder
+  @keys [:name, :popularity, :followers]
+  @enforce_keys @keys
+  defstruct @keys
+
+  def from_json(json) do
+    %Band{
+      name: json["name"],
+      popularity: json["popularity"],
+      followers: json["followers"]
+    }
+  end
+end
+
+defmodule Battle do
+  @keys [:left, :right]
+  @enforce_keys @keys
+  defstruct @keys
+end
```

Comment:
As a general comment, you would generally place these structs in something like `lib/emo_battle/band.ex` and `lib/emo_battle/battle.ex`. Where are you using these structs? In iex?